src/parsers.rs

```diff
@@ -182,7 +182,8 @@ impl<'a> Parser<'a> {
                 head[*head_idx] = part as u16;
                 *head_idx += 1;
 
-                if part == 0xffff && *head_idx == 6 {
+                if *head_idx == 6 && head[5] == 0xffff && head[4] == 0 && head[3] == 0
```

Comment:
nit: I like `head_idx == 6 && head[:head_idx] == [0, 0, 0, 0, 0, 0xffff]`

Reply:
i was not aware of that syntax. thanks

Reply:
I appear to have incorrectly incorporated python syntax in my example :)